bin/cli.js

```diff
@@ -1,17 +1,169 @@
+  'Opera': 'karma-opera-launcher',
+  'PhantomJS': 'karma-phantomjs-launcher',
+  'Safari': 'karma-safari-launcher'
+};
+
+const rmDuplicates = arr => arr.reduce((acc, cur) => {
+  if (acc.indexOf(cur) < 0) acc.push(cur);
+  return acc;
+}, []);
+
+const loadFiles = (files) => {
+  const result = [];
+  files.forEach(file => {
+    if (fs.statSync(file).isDirectory()) {
+      const subfiles = fs.readdirSync(file).map(f => `${file}/${f}`);
+      result.push(...loadFiles(subfiles));
+    } else if (file.indexOf('.js') > 0) {
+      result.push(file);
+    }
+  });
+  return result;
+};
+
+const setKarmaBrowsers = (config, ...browsers) => {
+  browsers = rmDuplicates(browsers);
+
+  if (!config.plugins) config.plugins = [];
+  if (!config.browsers) config.browsers = [];
+  browsers.forEach(browser => {
+    const launcher = browserLaunchers[browser];
+    if (launcher && !config.browsers[browser]) {
+      config.browsers.push(browser);
+      config.plugins.push(launcher);
+    }
+  });
+  return config;
+};
+
+const setKarmaLogLevel = (config, logLevel = 'disable') => {
+  if (!config.webpackMiddleware) config.webpackMiddleware = {};
+  config.logLevel = karma.constants[karmaLogLevels[logLevel][0]];
+  config.webpackMiddleware.stats = karmaLogLevels[logLevel][1];
+  return config;
+};
 
-walk('../test', (err, files) => {
-  metasync.for(files).each((file, next) => {
-    require(file.filePath);
-    next();
-  }).fetch(() => {
-    console.log('Tests done');
+const createFactory = () => {
+  const factory = function(files) {
+    files.unshift({ pattern: path.resolve(__dirname, '../karma/adapter.js') });
+  };
+  factory.$inject = ['config.files'];
+  return factory;
+};
+
+const getBrowserConfig = (conf) => {
+  const config = {
+    files: [],
+    preprocessors: {},
+    plugins: [
+      { 'framework:metatests': ['factory', createFactory()] },
+      'karma-webpack'
+    ],
+    concurrency: 1,
+    autoWatch: false,
+    singleRun: true,
+    frameworks: ['metatests'],
+    basePath: process.env.PWD,
+    webpack: { node: {} },
+    webpackMiddleware: {},
+    port: conf.browser.port
+  };
+  setKarmaBrowsers(config, ...conf.browser.browsers);
+  setKarmaLogLevel(config, conf.browser.logLevel);
+  conf.files.forEach(file => {
+    config.files.push({ pattern: file, type: 'module' });
+    config.preprocessors[file] = [ 'webpack' ];
+  });
+  [ 'fs', 'child_process' ].forEach(lib => {
+    config.webpack.node[lib] = 'empty';
   });
-});
+  return config;
+};
+
+const getConfig = () => {
+  const version = parseFile(path.resolve(__dirname, '../package.json')).version;
+  program.version(version).usage('[options] -- <file ...>');
+  clOptions.forEach(option => program.option(...option));
+  program.parse(process.argv);
+
+  const config = program.config ? parseFile(program.config) : {};
+
+  config.enviroment = program.enviroment;
+  config.files = config.files || [];
+  config.files.push(...program.args);
+  if (!config.files.length) {
```

Comment:
I'd say to always require a directory/files and just fail if there is none.

Reply:
If there won't be any files, node will exit with code 0 which I don't think should be the case and browser will just hang hang up.

Reply:
No, I meant to fail with an error if the user didn't provide files to run.

Reply:
This makes sense. Do you think I shouldn't try to load `test/all.js`?

Reply:
Yeah, that's what I meant. I don't think there should be some-implicit-file that will be loaded when the user didn't provide any, this is confusing at least.